api/lib/utilities/ccdDataStoreApiUtilities.ts

```diff
@@ -0,0 +1,90 @@
+import * as log4js from 'log4js'
+import {config} from '../../../config'
+import {asyncReturnOrError} from '../util'
+import {getEventTokenAndCase, postCaseWithEventToken} from '../../services/ccd-store-api/ccd-store'
+
+const logger = log4js.getLogger('dm-store')
+logger.level = config.logging || 'off'
+
+const ERROR_UNABLE_TO_GET_EVENT_TOKEN = 'Unable to retrieve Event Token. Required to start event creation as a case worker.'
+const ERROR_UNABLE_TO_POST_CASE = 'Unable to POST case data using the Event Token. Required to submit event creation as ' +
+    'case worker.'
+
+/**
+ * getEventToken
+ *
+ * An Event Token is required to update a case. This token needs to be sent through in the payload to update a case.
+ *
+ * The following is a generic function to get the Event Token.
+ *
+ * @see /caseworkers/{uid}/jurisdictions/{jid}/case-types/{ctid}/cases/{cid}/event-triggers/{etid}/token
+ * Which is used in the payload for:
+ * @see /caseworkers/{uid}/jurisdictions/{jid}/case-types/{ctid}/cases/{cid}/events
+ *
+ * @param userId
+ * @param caseId - '1548761902417900'
+ * @param {String} jurisdiction ie. 'DIVORCE'
+ * @param {String} caseType ie. 'FinancialRemedyMVP2'
+ * @param {String} eventId ie. 'FR_uploadDocument'
+ * @return {Promise}
+ */
+export async function getEventToken(userId, caseId, jurisdiction, caseType, eventId) {
+
+    try {
```

Comment:
could remove the try/catch and replace it with asyncReturnOrError?

Reply:
are these functions dupes of what exists in CCD-store?

Reply:
Replaced with asyncReturnOrError - done.
In regards to Dupes of what exists - the newly created functions are tested, and include error handling. But not to say that we can't bypass the newly created functions, but we would need to implement unit testing + error handling on the func in ccd-store. One to discuss in person*

Reply:
looking into it these functions basically wrap the ccd calls in a try catch, however this adds entire new files and directories for no reason. This could be done in the calling code. Similarly, the post method uses asyncReturn but then returns a rejected promise. It would be simpler and better to use ayncReturnOrError with ERROR_UNABLE_TO_POST_CASE on the code that calls this.

Then we get error handling without the unneeded layers.

Reply:
asyncReturnOrError usage. - Refactored Document Upload, Get Event Token, and Post Case
to use asyncReturnOrError completed.